core/iwasm/aot/aot_loader.c

```diff
@@ -1430,14 +1430,33 @@ load_types(const uint8 **p_buf, const uint8 *buf_end, AOTModule *module,
         }
         else if (type_flag == WASM_TYPE_STRUCT) {
             AOTStructType *struct_type;
-            uint16 field_count;
+            const uint8 *buf_org;
+            uint16 *reference_table;
+            uint16 field_count, ref_field_count = 0;
             uint32 offset;
+
             read_uint16(buf, buf_end, field_count);
             read_uint16(buf, buf_end, ref_type_map_count);
-            struct_type =
-                loader_malloc(sizeof(AOTStructType)
-                                  + field_count * sizeof(WASMStructFieldType),
-                              error_buf, error_buf_size);
+
+            buf_org = buf;
+
+            /* Traverse first time to get ref_field_count */
+            for (j = 0; j < field_count; j++) {
+                uint8 field_flags, field_type;
+
+                read_uint8(buf, buf_end, field_flags);
+                read_uint8(buf, buf_end, field_type);
+                if (wasm_is_type_reftype(field_type))
+                    ref_field_count++;
+
+                (void)field_flags;
+            }
+
+            struct_type = loader_malloc(
+                sizeof(AOTStructType)
+                    + sizeof(WASMStructFieldType) * (uint64)field_count
+                    + sizeof(uint16) * (uint64)(ref_field_count + 1),
```

Comment:
Why ref_field_count + 1, does it mean reserve one slot to store the `ref_field_count`?

Reply:
Yes, the first slot is to store the ref_field_count, see:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/dev/gc_refactor/core/iwasm/interpreter/wasm_loader.c#L1147-L1152
and
https://github.com/bytecodealliance/wasm-micro-runtime/blob/dev/gc_refactor/core/iwasm/common/gc/gc_object.c#L737-L743

Reply:
Got it, thanks!